### DISCUSSION TITLE
My website is still not indexed properly on Google

### DISCUSSION THREAD

**Discussion by kim-lindgren** [gatsbyjs/gatsby#37681] [Help] [answered]
I decided to build my website using Gatsby because I heard that it was search engine-friendly. The blog posts are built as .md or .mdx files, which supposedly should make them more approachable to search engine bots than something like React.

After almost a year and some 30-40 visitors a month, my site is still not properly indexed. The following pages are indexed:
- The landing page (https://plutoastrology.net/)
- Counseling
- Reviews
- Blog (https://plutoastrology.net/blog - only provides an overview of the blog posts) 

The following are not at all searchable, even when I paste whole sentences into Google and search for them:
- Contact
- About
- ANY individual blog post! This last one really is a bit of a problem.

I am a simple frontend developer and I don't know much about sitemaps, caching or generally how Gatsby sets everything up. I just thought it would work. Please help!

**Comment by gh640** [comment, marked as answer]
As you know, as it depends on the context and many factors, it's difficult to say what the cause is here.

I don't know your context but it makes sense to me that not all pages are indexed if there are 30-40 visitors a month. There are cases Google doesn't index some pages in a site even if there're a few thousands of visitors a month. Google assesses pages and decides if it's worth to index.

In your case, I guess the cause is not by Gatsby as some pages are already indexed.

I recommend to read Google's official explanation about indexing. For example:

- https://developers.google.com/search/docs/fundamentals/get-on-google?hl=en
- https://support.google.com/programmable-search/answer/4513925?hl=en
- https://developers.google.com/search/docs/crawling-indexing?hl=en

**Reply by kim-lindgren** [reply]
Thanks for your suggestion. Perhaps I overestimated Google's willingness to crawl my website.

Just a thought, could it have an impact that I customized my blog directories? Right now all blog posts are sorted in /blog/2021 or blog/2022, so one folder for every year. Perhaps Google goes to blog/2021 and then doesn't go any further because it can't find anything there. I understand if you don't know the answer to this... in any case, thanks for pointing this out.

**Reply by gh640** [reply]
Thank you for your reply.

I don't think removing the year folder in the post paths (URLs) effects how Google crawls and indexes pages. If my understanding is correct, Google basically finds new pages by using links in known pages not by digging path folders.

If you have not used Google Search Console, I recommend to use it for your site. Search Console lets you know how Google recognizes your site. You might get advices to improve your site from Google's point of view.

- https://search.google.com/search-console/about

**Comment by 808vita** [comment]
adding comment even thought it's an old post .

![image](https://user-images.githubusercontent.com/97225946/231912030-f336736f-94fe-45e0-a012-8999f87dce83.png)

Only three urls are indexed to google (that's what it displays ) .
So yes not all of your blog articles are indexed .

 What you seek for would come under search engine optimization (seo) . You can ping crawler (aka inform google to crawl links) using third party sites but it is better to create google search console account and index using it .

I will try to provide some instructions here -

- create /signup to google search console 
- add your property (your site)
- create an sitemap.xml file - make use online sitemap generator or sitemap plugin - add it your site - make sure it contains all the required urls* https://www.gatsbyjs.com/plugins/gatsby-plugin-sitemap/

 or  ( alternate way - manual but good enough for low number of urls) 
- make a list of all your site links , everything that needs to be indexed and manually use "url inspection" in search console and select "request to index ".

It might take some to take effect but if its being shown in the search console without any errors you are good . *The ranking & traffic would depend on the content on site 
 and competition from other competing sites.
